filebeat/beater/filebeat.go

```diff
@@ -48,10 +51,10 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 	var wgEvents *sync.WaitGroup // count active events for waiting on shutdown
 	var finishedLogger publisher.SuccessLogger
 
-	if fb.config.ShutdownTimeout > 0 {
-		wgEvents = &sync.WaitGroup{}
-		finishedLogger = newFinishedLogger(wgEvents)
-	}
+	//if fb.config.ShutdownTimeout > 0 || *once {
```

Comment:
Will leave it in as default for the moment and we can still make it optional in case it becomes a performance issue (I would not assume so)